tests/test_core.py

```diff
@@ -955,6 +956,31 @@ def test_is_allowed_path(self):
             with pytest.raises(AssertionError):
                 self.config.is_allowed_path(None)
 
+    def test_is_allowed_external_url(self):
+        """Test is_allowed_external_url method."""
+        self.config.allowlist_external_urls = [
+            "http://x.com/",
+            "https://y.com/bla/",
+            "https://z.com/images/1.jpg/",
+        ]
+
+        valid = [
+            "http://x.com/1.jpg",
+            "https://y.com/bla/2.png",
```

Comment:
Let's add these two cases which should also be valid.
```suggestion
            "http://x.com",
            "https://y.com/bla/",
            "https://y.com/bla/2.png",
```